### PR TITLE
fix nullable reflected references

### PR DESCRIPTION
**Pull request by jtenner**
We should assert that `left` and `right` values, when compared, are not nullable by the time the flow hits the `instanceof` checks.

### CODE REVIEW

File: packages/assembly/assembly/internal/Reflect.ts [hunk 1]
```diff
@@ -454,7 +458,7 @@ export class Reflect {
       if (isDefined(left[0])) {
         // test for safe indexof usage
         // set match
-        if (left instanceof Set) {
+        if (left! instanceof Set) {
```
Comment (DuncanUszkay1) [inline packages/assembly/assembly/internal/Reflect.ts]: why do we need to assert here again? Shouldn't line 412 already guarantee left is not nullable?

File: packages/assembly/assembly/internal/Reflect.ts [hunk 2]
```diff
@@ -397,23 +397,22 @@ export class Reflect {
...
-      return Reflect.FAILED_MATCH;
-    }
-
-    // if it's possible for T to be null
-    if (isNullable<T>()) {
-      // mutual exclusion null
-      if (
-        i32(changetype<usize>(left) == 0) ^ i32(changetype<usize>(right) == 0)
-      )
+    if (isReference(left)) {
```
Comment (willemneal) [inline packages/assembly/assembly/internal/Reflect.ts]: Why not `isReference<T>()`?

File: packages/assembly/assembly/internal/Reflect.ts (hunk 1 above)
Comment (jtenner) [reply packages/assembly/assembly/internal/Reflect.ts]: Yeah that's part of the flow error I've found. Not sure how to fix it.

File: packages/assembly/assembly/internal/Reflect.ts (hunk 2 above)
Comment (jtenner) [reply packages/assembly/assembly/internal/Reflect.ts]: I don't mind relying on type inference here.

Comment (willemneal) [reply packages/assembly/assembly/internal/Reflect.ts]: Ah well since you did, `isNullable<T>()` above, we should choose one or the other for consistency.

Comment (jtenner) [reply packages/assembly/assembly/internal/Reflect.ts]: fair enough. I'll just change it.

**Pull request merged**